lib/phoenix/controller.ex

```diff
@@ -925,13 +926,16 @@ defmodule Phoenix.Controller do
 
   defp prepare_send_download(conn, filename, opts) do
     content_type = opts[:content_type] || MIME.from_path(filename)
-    encoded_filename = URI.encode_www_form(filename)
+    encoded_filename = encode_filename(filename, opts[:encode])
     warn_if_ajax(conn)
     conn
     |> put_resp_content_type(content_type, opts[:charset])
     |> put_resp_header("content-disposition", ~s[attachment; filename="#{encoded_filename}"])
   end
 
+  defp encode_filename(filename, encode) when encode == false, do: filename
+  defp encode_filename(filename, encode), do: URI.encode_www_form(filename)
```

Comment:
better to explicitly mark unused variables
`encode` -> `_encode`

Reply:
nice. I changed it.